### PR TITLE
[query] Fix bug in call decoding, add error ID to lgt_to_gt

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/main/scala/is/hail/variant/Call.scala [hunk 1]
```diff
@@ -43,6 +43,19 @@ object Call2 {
 
     Call(alleleRepr, phased, ploidy = 2)
   }
+
+  def withErrorID(aj: Int, ak: Int, phased: Boolean, errorID: Int): Call = {
+    if (aj < 0 || ak < 0)
+      fatal(s"allele indices must be >= 0. Found j=$aj and k=$ak.")
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/variant/Call.scala:49]: Should this be tagged with the errorID too?

Comment (tpoterba) [reply hail/src/main/scala/is/hail/variant/Call.scala:49]: yep, good catch

**Comment by johnc1231** [comment]
Thanks for catching this, sorry about that.

**Comment by johnc1231** [comment]
@tpoterba failing the test you wrote

**Comment by tpoterba** [comment]
yep, will fix when done with docs

**Comment by tpoterba** [comment]
Commented out the phased test -- this was meant to fail, and I didn't intend to fix that in this PR (it's hard, we need to change the Call representation...)

**Comment by tpoterba** [comment]
Addressed Patrick's comment.

**Review by patrick-schultz — APPROVED** [review]

**Pull request merged**